### PR TITLE
Fix Microsoft.Extensions.DependencyModel AssemblyVersionInPackageVersion

### PR DESCRIPTION
**Pull request by eerhardt**

### CODE REVIEW

**Review by joperezr — APPROVED** [review]
Thanks for fixing

**Review by Anipik — APPROVED** [review]

File: src/libraries/pkg/baseline/packageIndex.json
Comment (Anipik) [inline src/libraries/pkg/baseline/packageIndex.json:553]: dont we increment the last part in the servicing releases ?

**Comment by stephentoub** [comment]
Can we merge this to unblock PRs?

**Comment by eerhardt** [comment]
> Can we merge this to unblock PRs?

Nothing should be blocked on this. My previous change fixed those.  https://github.com/dotnet/runtime/pull/36285  This is just a follow up to that.

File: src/libraries/pkg/baseline/packageIndex.json
Comment (joperezr) [reply src/libraries/pkg/baseline/packageIndex.json:553]: Maybe but this is just reacting to a package serviced in release/3.1 which was not yet part of libraries so may use different versioning

**Comment by stephentoub** [comment]
Ah, ok, then we probably need to close and re-open some PRs to pick up your previous change.

**Comment by joperezr** [comment]
As @eerhardt suggests PRs shouldn’t be block by this, that said I’ll go ahead and merge as we don’t have anything else to address here

File: src/libraries/pkg/baseline/packageIndex.json
Comment (eerhardt) [reply src/libraries/pkg/baseline/packageIndex.json:553]: I’m not sure I understand the question.

This library was not part of corefx. This is the first release it is being built using the “corefx/libraries” packaging infrastructure.

Comment (joperezr) [reply src/libraries/pkg/baseline/packageIndex.json:553]: He is saying that usually in servicing we only increment  the last part of the assembly version, so package 3.1.4 should have had assembly version 3.1.3.1 instead of 3.1.4.0, but that wasn’t a decision made on this PR, this is just reacting to what already happened in servicing

Comment (Anipik) [reply src/libraries/pkg/baseline/packageIndex.json:553]: yes i should have been more clearer, i meant to ask if the versioning scheme used here is on purpose or just a mistake in packaging (nothing wrong with this pr)

Comment (joperezr) [reply src/libraries/pkg/baseline/packageIndex.json:553]: @Anipik keep in mind that at least in release/3.1 this package is being built on a different repo (coresetup) which means that they were using a slightly different infrastructure and were possibly not following the same versioning guidelines we do in corefx. Now we have moved it to runtime it should start following conventions starting from 5.0 release

**Pull request merged**
